tls/extensions/s2n_client_key_share.c

```diff
@@ -76,7 +76,9 @@ int s2n_extensions_client_key_share_recv(struct s2n_connection *conn, struct s2n
     uint16_t named_group, share_size;
     int supported_curve_index;
```

Comment:
any reason to leave this and not make it explicit too?

Reply:
The number of supported curves currently is 2, so no matter what size int takes it won't overflow since it can only take on the values of 0, 1, and 2 during runtime, and we're unlikely to add 2^16 individual supported curves in the future, which would be the only way to run up against an overflow.

Reply:
Still, while you're in, might as well fix it.

Reply:
Sure, done and future-proofed it, doubt we'll hit 2^32 supported curves any time soon